src/components/rootcomponents/LoginInput.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+interface InputProps {
+  id: string;
+  type: string;
+  placeholder: string;
+  register: any;
+  errors: any;
+}
+
+const Input: React.FC<InputProps> = ({ id, type, placeholder, register, errors }) => {
+  return (
+    <div className="mb-4">
+      <input
+        {...register(id)}
+        id={id}
+        type={type}
+        placeholder={placeholder}
+        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+      />
+      {errors[id] && typeof errors[id].message === 'string' && (
+        <p className="text-red-500 text-md">{errors[id].message}</p>
+      )}
+    </div>
+  );
+};
+
+export default Input;
```

Comment:
Can't we have one input which can be reusable everywhere? Why make another component here when we already had it?